wpimath/generate_numbers.py

```diff
@@ -11,23 +11,32 @@
     template = templateFile.read()
     rootPath = f"{cmake_binary_dir}/generated/main/java/edu/wpi/first/wpiutil/math/numbers"
 
+    doWrite = True
     if os.path.exists(rootPath):
-        shutil.rmtree(rootPath)
-    os.makedirs(rootPath)
-
-    for i in range(MAX_NUM + 1):
-        with open(f"{rootPath}/N{i}.java", "w") as f:
-            f.write(template.replace("${num}", str(i)))
+        count = 0
+        for i in range(MAX_NUM + 1):
+            if os.path.exists(f"{rootPath}/N{i}.java"):
+                with open(f"{rootPath}/N{i}.java", "r") as f:
+                    if f.read() == template.replace("${num}", str(i)):
+                        count += 1
+        if count == MAX_NUM + 1:
+            doWrite = False
```

Comment:
Could do a break and for-else instead here?

Reply:
Tyler rewrote it.